File: ProyectosJAVA/lp2softRRHHController/src/pe/edu/pucp/lp2soft/rrhh/mysql/EmpleadoMySQL.java

package pe.edu.pucp.lp2soft.rrhh.mysql;

import java.sql.CallableStatement;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.Statement;
import java.text.SimpleDateFormat;
import java.util.ArrayList;
import pe.edu.pucp.lp2soft.rrhh.config.DBManager;
import pe.edu.pucp.lp2soft.rrhh.dao.EmpleadoDAO;
import pe.edu.pucp.lp2soft.rrhh.model.Empleado;
import java.util.Date;
import pe.edu.pucp.lp2soft.rrhh.model.Area;
public class EmpleadoMySQL implements EmpleadoDAO{
    Connection con; //java.sql.Connection
    Statement st; //java.sql.Statement
    ResultSet rs;
    CallableStatement cs;
    @Override
    public ArrayList<Empleado> listar() {
        ArrayList<Empleado> empleados = new ArrayList<>();
        try{
            Class.forName("com.mysql.cj.jdbc.Driver");
            con = DriverManager.getConnection(DBManager.urlMySQL, DBManager.user, DBManager.password);
            cs = con.prepareCall("{call LISTAR_EMPLEADOS()}");
            rs = cs.executeQuery();
            while(rs.next()){
                Empleado empleado = new Empleado();
                empleado.setIdPersona(rs.getInt("id_persona"));
                empleado.setNombreCompleto(rs.getString("nombre_completo"));
                empleado.setDNI(rs.getString("DNI"));
                empleado.setFechaNacimiento(rs.getDate("fecha_nacimiento"));
                empleado.setEdad(rs.getInt("edad"));
                empleado.setGenero(rs.getString("genero").charAt(0));
                empleado.setCargo(rs.getString("cargo"));
                empleado.setSueldo(rs.getDouble("sueldo"));
                empleado.setArea(new Area());
                empleado.getArea().setIdArea(rs.getInt("id_area"));
                empleado.getArea().setNombre(rs.getString("nombre_area"));
                empleado.setActivo(true);
                empleados.add(empleado);
            }
            rs.close();
            cs.close();
        }catch(Exception ex){
            System.out.println(ex.getMessage());
        }finally{
            try{con.close();}catch(Exception ex){System.out.println(ex.getMessage());}
        }
        return empleados;
    }

    @Override
    public int insertar(Empleado empleado) {
        int resultado=0;
        try{
            //registrar el driver
           Class.forName("com.mysql.cj.jdbc.Driver");
           //creamos la conexion
           con = DriverManager.getConnection(DBManager.urlMySQL,
                   DBManager.user,DBManager.password);
           //creamos un statement para ejecutar una instruccion
           cs = con.prepareCall("{call INSERTAR_EMPLEADO(?,?,?,?,?,?,?,?,?)}");
           cs.registerOutParameter("_id_empleado",java.sql.Types.INTEGER);
           cs.setInt("_fid_area",empleado.getArea().getIdArea());
           cs.setString("_nombre_completo",empleado.getNombreCompleto());
           cs.setString("_DNI",empleado.getDNI());
           cs.setDate("_fecha_nacimiento",new java.sql.Date(empleado.getFechaNacimiento().getTime()));
           cs.setInt("_edad", empleado.getEdad());
           cs.setString("_genero", String.valueOf(empleado.getGenero()));
           cs.setString("_cargo", empleado.getCargo());
           cs.setDouble("_sueldo", empleado.getSueldo());
           cs.executeUpdate();
           empleado.setIdPersona(cs.getInt("_id_empleado"));
           resultado=1;
           cs.close();
        }catch(Exception ex){
            System.out.println(ex.getMessage());
        }finally{
            //esto siempre se va a ejecutar
            try{con.close();}catch(Exception ex){System.out.println(ex.getMessage());}
        }
        
        return resultado;
    }

    @Override
    public int modificar(Empleado empleado) {
       int resultado = 0;
        try{
            Class.forName("com.mysql.cj.jdbc.Driver");
            con = DriverManager.getConnection(DBManager.urlMySQL, DBManager.user, DBManager.password);
            cs = con.prepareCall("{call MODIFICAR_EMPLEADO(?,?,?,?,?,?,?,?,?)}");
            cs.setInt("_id_empleado", empleado.getIdPersona());
            cs.setInt("_fid_area", empleado.getArea().getIdArea());
            cs.setString("_nombre_completo", empleado.getNombreCompleto());
            cs.setString("_DNI", empleado.getDNI());
            cs.setDate("_fecha_nacimiento", new java.sql.Date(empleado.getFechaNacimiento().getTime()));
            cs.setInt("_edad", empleado.getEdad());
            cs.setString("_genero", String.valueOf(empleado.getGenero()));
            cs.setString("_cargo", empleado.getCargo());
            cs.setDouble("_sueldo", empleado.getSueldo());
            cs.executeUpdate();
            resultado = 1;
            cs.close();
        }catch(Exception ex){
            System.out.println(ex.getMessage());
        }finally{
            try{con.close();}catch(Exception ex){System.out.println(ex.getMessage());};
        }
        return resultado;
    }

    @Override
    public int eliminar(int idEmpleado) {
        int resultado = 0;
        try{
            Class.forName("com.mysql.cj.jdbc.Driver");
            con = DriverManager.getConnection(DBManager.urlMySQL, DBManager.user, DBManager.password);
            cs = con.prepareCall("{call ELIMINAR_EMPLEADO(?)}");
            cs.setInt("_id_empleado", idEmpleado);
            cs.executeUpdate();
            resultado = 1;
            cs.close();
        }catch(Exception ex){
            System.out.println(ex.getMessage());
        }finally{
            try{con.close();}catch(Exception ex){System.out.println(ex.getMessage());};
        }
        return resultado;
    }
    
}

File: ProyectosNET/LP2soft/LP2softRRHHController/MySQL/EmpleadoMySQL.cs
using LP2softRRHHController.Config;
using LP2softRRHHController.DAO;
using LP2softRRHHModel;
using MySql.Data.MySqlClient;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LP2softRRHHController.MySQL
{
    public class EmpleadoMySQL : EmpleadoDAO
    {
        MySqlConnection con;
        MySqlCommand comando;
        MySqlDataReader lector;
        public int eliminar(int idEmpleado)
        {
            throw new NotImplementedException();
        }

        public int insertar(Empleado empleado)
        {
            int resultado = 0;
            try
            {
                con = new MySqlConnection(DBManager.cadCon);
                con.Open();
               
                comando = new MySqlCommand();
                comando.Connection = con;
                comando.CommandType = System.Data.CommandType.StoredProcedure;
                comando.CommandText = "INSERTAR_EMPLEADO";

                comando.Parameters.Add("_id_empleado", MySqlDbType.Int32).Direction
                    = System.Data.ParameterDirection.Output;
                comando.Parameters.AddWithValue("_fid_area", empleado.Area.IdArea);
                comando.Parameters.AddWithValue("_nombre_completo", empleado.NombreCompleto);
                comando.Parameters.AddWithValue("_DNI", empleado.DNI);
                comando.Parameters.AddWithValue("_fecha_nacimiento", empleado.FechaNacimiento);
                comando.Parameters.AddWithValue("_edad", empleado.Edad);
                comando.Parameters.AddWithValue("_cargo", empleado.Cargo);
                comando.Parameters.AddWithValue("_genero", empleado.Genero);
                comando.Parameters.AddWithValue("_sueldo", empleado.Sueldo);
                comando.ExecuteNonQuery();

                empleado.IdPersona = Int32.Parse(comando.Parameters["_id_empleado"].Value.ToString());
                resultado = 1;
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message);
            }
            finally
            {
                con.Close();
            }

            return resultado;
        }

        public BindingList<Empleado> listar()
        {
            BindingList<Empleado> empleados = new BindingList<Empleado>();
            try
            {
                con = new MySqlConnection(DBManager.cadCon);
                con.Open();
                comando = new MySqlCommand();
                comando.Connection = con;
                comando.CommandType = System.Data.CommandType.StoredProcedure;
                comando.CommandText = "LISTAR_EMPLEADOS";
                lector = comando.ExecuteReader();

                while (lector.Read())
                {
                    Empleado empleado = new Empleado();
                    empleado.IdPersona = lector.GetInt32("id_persona");
                    empleado.NombreCompleto = lector.GetString("nombre_completo");
                    empleado.DNI = lector.GetString("DNI");
                    empleado.FechaNacimiento = lector.GetDateTime("fecha_nacimiento");
                    empleado.Edad = lector.GetInt32("edad");
                    empleado.Genero = lector.GetChar("genero");
                    empleado.Cargo = lector.GetString("cargo");
                    empleado.Sueldo = lector.GetDouble("sueldo");
                    empleado.Area = new Area();
                    empleado.Area.IdArea = lector.GetInt32("id_area");
                    empleado.Area.Nombre = lector.GetString("nombre_area");
                    empleado.Area.Activo = true;

                    empleados.Add(empleado);
                }

                lector.Close();
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message);
            }
            finally
            {
                con.Close();
            }

            return empleados;
        }

        public int modificar(Empleado empleado)
        {
            throw new NotImplementedException();
        }
    }
}
